parser/src/ast/trace.rs

```diff
@@ -230,6 +230,8 @@ pub struct TraceBinding {
     pub size: usize,
     /// The effective type of this binding
     pub ty: Type,
+    /// The access type associated to this TraceBinding
+    pub access: Option<AccessType>,
```

Comment:
What this filed mean here? That is, in what situations would a trace binding have `access` vs. when it could be `None`?